Source/Scene/Label.js

```diff
@@ -1196,5 +1246,171 @@ define([
         return false;
     };
 
+    function declareTypes() {
+        var TextTypes = {
+            LTR : 0,
+            RTL : 1,
+            WEAK : 2,
+            BRACKETS : 3
+        };
+        return freezeObject(TextTypes);
+    }
+
+    function convertTextToTypes(text, rtlDir, rtlChars) {
+        var ltrChars = /[a-zA-Z0-9]/;
+        var bracketsChars = /[()[\]{}<>]/;
+        var parsedText = [];
+        var word = '';
+        var types = declareTypes();
+        var lastType = rtlDir ? types.RTL : types.LTR;
+        var currentType = '';
+        var textLength = text.length;
+        for (var textIndex = 0; textIndex < textLength; ++textIndex) {
+            var character = text.charAt(textIndex);
+            if (rtlChars.test(character)) {
+                currentType = types.RTL;
+            }
+            else if (ltrChars.test(character)) {
+                currentType = types.LTR;
+            }
+            else if (bracketsChars.test(character)) {
+                currentType = types.BRACKETS;
+            }
+            else {
+                currentType = types.WEAK;
+            }
+
+            if (lastType === currentType && currentType !== types.BRACKETS) {
+                word += character;
+            }
+            else {
+                parsedText.push({Type : lastType, Word : word});
+                lastType = currentType;
+                word = character;
+            }
+        }
+        parsedText.push({Type : currentType, Word : word});
+        return parsedText;
+    }
+
+    function reverseWord(word) {
+        return word.split('').reverse().join('');
+    }
+
+    function spliceWord(result, pointer, word) {
+        return result.slice(0, pointer) + word + result.slice(pointer);
+    }
+
+    function reverseBrackets(bracket) {
+        switch(bracket) {
+            case '(':
+                return ')';
+            case ')':
+                return '(';
+            case '[':
+                return ']';
+            case ']':
+                return '[';
+            case '{':
+                return '}';
+            case '}':
+                return '{';
+            case '<':
+                return '>';
+            case '>':
+                return '<';
+        }
+    }
+
+    /**
+     *
+     * @param {String} text the text to parse and reorder
+     * @returns {String} the text as rtl direction
+     */
+    Label.prototype.reverseRtl = function(value) {
```

Comment:
This function should be private, I don't think anyone should be calling this directly.